compiler/rustc_hir/src/def.rs

```diff
@@ -275,17 +379,23 @@ impl PartialRes {
     }
 }
 
-/// Different kinds of symbols don't influence each other.
-///
-/// Therefore, they have a separate universe (namespace).
+/// Different kinds of symbols can coexist even if they share the same textual name.
+/// Therefore, they each have a separate universe (known as a "namespace").
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
 pub enum Namespace {
+    /// The type namespace includes `struct`s, `enum`s, `union`s, `trait`s, and `mod`s
+    /// (and, by extension, crates).
     TypeNS,
+    /// The value namespace includes `fn`s, `const`s, and `static`s.
```

Comment:
```suggestion
    /// The value namespace includes `fn`s, `const`s, `static`s, and local variables.
```

Reply:
(added a note to clarify things a bit)

```suggestion
    /// The value namespace includes `fn`s, `const`s, `static`s, and local variables (including function arguments).
```